servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcClientBuilder.java

```diff
@@ -50,72 +58,235 @@
 
-    @Override
+    /**
+     * Updates the automatic retry strategy for the clients generated by this builder. Automatic retries are done by
+     * the clients automatically when allowed by the passed {@link AutoRetryStrategyProvider}. These retries are not a
+     * substitute for user level retries which are designed to infer retry decisions based on request/error information.
+     * Typically such user level retries are done using filters but can also be done differently per request
+     * (eg: by using {@link Single#retry(BiIntPredicate)}).
+     *
+     * @param autoRetryStrategyProvider {@link AutoRetryStrategyProvider} for the automatic retry strategy.
+     * @return {@code this}
+     */
     public abstract GrpcClientBuilder<U, R> autoRetryStrategy(
             AutoRetryStrategyProvider autoRetryStrategyProvider);
 
-    @Override
+    /**
+     * Provides a means to convert {@link U} unresolved address type into a {@link CharSequence}.
+     * An example of where this maybe used is to convert the {@link U} to a default host header. It may also
+     * be used in the event of proxying.
+     *
+     * @param unresolvedAddressToHostFunction invoked to convert the {@link U} unresolved address type into a
+     * {@link CharSequence} suitable for use in
+     * <a href="https://tools.ietf.org/html/rfc7230#section-5.4">Host Header</a> format.
+     * @return {@code this}
+     */
     public abstract GrpcClientBuilder<U, R> unresolvedAddressToHost(
             Function<U, CharSequence> unresolvedAddressToHostFunction);
 
-    @Override
+    /**
+     * Disables automatically setting {@code Host} headers by inferring from the address or {@link HttpMetaData}.
+     * <p>
+     * This setting disables the default filter such that no {@code Host} header will be manipulated.
+     *
+     * @return {@code this}
+     * @see #unresolvedAddressToHost(Function)
+     */
     public abstract GrpcClientBuilder<U, R> disableHostHeaderFallback();
 
-    @Override
+    /**
+     * Set a {@link ServiceDiscoverer} to resolve addresses of remote servers to connect to.
+     * @param serviceDiscoverer The {@link ServiceDiscoverer} to resolve addresses of remote servers to connect to.
+     * Lifecycle of the provided {@link ServiceDiscoverer} is managed externally and it should be
+     * {@link ServiceDiscoverer#closeAsync() closed} after all built {@link GrpcClient}s are closed and
+     * this {@link ServiceDiscoverer} is no longer needed.
+     * @return {@code this}.
+     */
     public abstract GrpcClientBuilder<U, R> serviceDiscoverer(
             ServiceDiscoverer<U, R, ServiceDiscovererEvent<R>> serviceDiscoverer);
 
-    @Override
+    /**
+     * Set a {@link HttpLoadBalancerFactory} to create {@link LoadBalancer} instances.
+     *
+     * @param loadBalancerFactory {@link HttpLoadBalancerFactory} to create {@link LoadBalancer} instances.
+     * @return {@code this}.
+     */
```

Comment:
Can not comment on line 381:
Interesting that we do not have any test or example for `buildMulti()`. We should revisit if it's useful or not.